android/src/com/mapswithme/maps/downloader/MigrationController.java

```diff
@@ -149,7 +149,7 @@ void start(boolean keepOld)
     if (mState == State.PROGRESS)
       return;
 
-    Location loc = LocationHelper.INSTANCE.getLastLocation();
+    Location loc = LocationHelper.INSTANCE.getLastKnownLocation();
```

Comment:
Я именно такой вариант и имел ввиду при обсуждении решения этой задачи.

> On Apr 1, 2016, at 13:52, Dmitry Yunitsky <notifications@github.com> wrote:
> 
> In android/src/com/mapswithme/maps/downloader/MigrationController.java <https://github.com/mapsme/omim/pull/2672#discussion_r58190359>:
> 
> > @@ -149,7 +149,7 @@ void start(boolean keepOld)
> >      if (mState == State.PROGRESS)
> >        return;
> >  
> > -    Location loc = LocationHelper.INSTANCE.getLastLocation();
> > +    Location loc = LocationHelper.INSTANCE.getLastKnownLocation();
> Для этого можно просто дёргать системный LocationManager.getLastKnownLocation(LocationManager.PASSIVE_PROVIDER); Вообще и здесь тоже можно упростить и позицию получать из него, не перебирая все провайдеры.
> … <x-msg://42/#>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> 
>  <https://github.com/mapsme/omim/pull/2672#discussion_r58189614> <https://github.com/mapsme/omim/pull/2672#discussion_r58189451> <https://github.com/mapsme/omim/pull/2672/files/cc92bbc4b790b789ab9d08f4d3d2825564b6054a#r58189451> <https://github.com/mapsme/omim/pull/2672/files/cc92bbc4b790b789ab9d08f4d3d2825564b6054a#r58189614>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2672/files/cc92bbc4b790b789ab9d08f4d3d2825564b6054a#r58190359>